shared/slotutil/slotticker.go

```diff
@@ -51,6 +51,23 @@ func GetSlotTicker(genesisTime time.Time, secondsPerSlot uint64) *SlotTicker {
 	return ticker
 }
 
+// GetSlotTickerWithOffset is a constructor for SlotTicker that allows a offset of time from genesis,
+// entering a offset greater than secondsPerSlot is not allowed.
+func GetSlotTickerWithOffset(genesisTime time.Time, offset time.Duration, secondsPerSlot uint64) *SlotTicker {
+	if genesisTime.Unix() == 0 {
+		panic("zero genesis time")
+	}
+	if offset > time.Duration(secondsPerSlot)*time.Second {
+		panic("invalid ticker offset")
```

Comment:
Just to double check. Panic here is the best course of action?

Reply:
We do this in the slot ticker today, maybe we can amend the regular slot ticker to not panic as well?

Reply:
Yeah, wasn't sure how to handle it since we panic for existing cases, could change the ticker to return an error in a diff PR, or just change the offset ticker to return an error for now. Wdyt?

Reply:
Im down to do it in a separate PR, thoughts @terencechain ?